src/analysis/special_functions/gaussian.lean

```diff
@@ -511,26 +518,94 @@ begin
-  ∫ (x : ℝ), cexp (2 * π * I * t * x) * cexp (-π * b * x ^ 2) =
-    1 / b ^ (1 / 2 : ℂ) * cexp (-π * (1 / b) * t ^ 2) :=
+lemma _root_.fourier_transform_gaussian_pi (hb : 0 < b.re) :
+  𝓕 (λ x : ℝ, cexp (-π * b * x ^ 2)) = λ t : ℝ, 1 / b ^ (1 / 2 : ℂ) * cexp (-π / b * t ^ 2) :=
 begin
+  ext1 t,
+  simp_rw [fourier_integral_eq_integral_exp_smul, smul_eq_mul],
   have h1 : 0 < re (π * b) := by { rw of_real_mul_re, exact mul_pos pi_pos hb },
   have h2 : b ≠ 0 := by { contrapose! hb, rw [hb, zero_re], },
-  convert _root_.fourier_transform_gaussian h1 (2 * π * t) using 1,
-  { congr' 1,
-    ext1 x,
+  convert _root_.fourier_transform_gaussian h1 (-2 * π * t) using 1,
+  { congr' 1 with x:1,
     congr' 2,
-    all_goals { ring } },
+    all_goals { push_cast, ring } },
   { conv_lhs { rw mul_comm },
     congr' 2,
     { field_simp [of_real_ne_zero.mpr pi_ne_zero], ring, },
-    { rw [←div_div, div_self (of_real_ne_zero.mpr pi_ne_zero), cpow_def_of_ne_zero h2,
-        cpow_def_of_ne_zero (one_div_ne_zero h2), one_div, ←complex.exp_neg, ←neg_mul],
-      congr' 2,
-      rw [one_div, complex.log_inv],
+    { rw [←div_div, div_self (of_real_ne_zero.mpr pi_ne_zero), one_div, one_div b, inv_cpow],
       rw [ne.def, arg_eq_pi_iff, not_and_distrib, not_lt],
       exact or.inl hb.le } },
 end
 
 end gaussian_fourier
+
+section gaussian_poisson
+/-! ## Poisson summation applied to the Gaussian -/
+
+variables {E : Type*} [normed_add_comm_group E]
+
+lemma tendsto_rpow_abs_mul_exp_neg_mul_sq_cocompact {a : ℝ} (ha : 0 < a) (s : ℝ) :
+  tendsto (λ x : ℝ, |x| ^ s * rexp (-a * x ^ 2)) (cocompact ℝ) (𝓝 0) :=
+begin
+  conv in (rexp _) { rw ←sq_abs },
+  rw [cocompact_eq, ←comap_abs_at_top,
+    @tendsto_comap'_iff _ _ _ (λ y, y ^ s * rexp (-a * y ^ 2)) _ _ _
+    (mem_at_top_sets.mpr ⟨0, λ b hb, ⟨b, abs_of_nonneg hb⟩⟩)],
+  exact (rpow_mul_exp_neg_mul_sq_is_o_exp_neg ha s).tendsto_zero_of_tendsto
+    (tendsto_exp_at_bot.comp $ tendsto_id.neg_const_mul_at_top (neg_lt_zero.mpr one_half_pos)),
+end
+
+lemma is_o_exp_neg_mul_sq_cocompact {a : ℂ} (ha : 0 < a.re) (s : ℝ) :
+  (λ x : ℝ, complex.exp (-a * x ^ 2)) =o[cocompact ℝ] (λ x : ℝ, |x| ^ s) :=
+begin
+  rw ←is_o_norm_left,
+  simp_rw norm_cexp_neg_mul_sq,
+  apply is_o_of_tendsto',
+  { refine eventually.filter_mono cocompact_le_cofinite _,
+    refine (eventually_cofinite_ne 0).mp (eventually_of_forall (λ x hx h, _)),
+    exact ((rpow_pos_of_pos (abs_pos.mpr hx) _).ne' h).elim },
+  { refine tendsto.congr' (eventually.filter_mono cocompact_le_cofinite _)
+      (tendsto_zero_iff_norm_tendsto_zero.mp $
+      tendsto_rpow_abs_mul_exp_neg_mul_sq_cocompact ha (-s)),
+    refine (eventually_cofinite_ne 0).mp (eventually_of_forall (λ x hx, _)),
+    rw [norm_mul, norm_of_nonneg (rpow_nonneg_of_nonneg (abs_nonneg _) _), mul_comm,
+      rpow_neg (abs_nonneg x), div_eq_mul_inv, norm_of_nonneg (exp_pos _).le] },
+end
+
+lemma complex.tsum_exp_neg_mul_int_sq {a : ℂ} (ha : 0 < a.re) :
```

Comment:
This one is probably true without `ha`, for the silly reason that non-summable series have zero sum.

Reply:
Yes, that is true, but it is a statement of no mathematical interest whatsoever, and it will probably take a couple of dozen lines to prove. This lemma's not going to be invoked 100's of times -- I'm adding it because it's a prerequisite for one _big_ theorem, not hundreds of little theorems. So I'm not convinced that the slight labour-saving when calling this theorem would be worth the extra effort in the proof to drop `ha`.

Reply:
ok